services/datamanager/builtin/builtin.go

```diff
@@ -539,30 +544,33 @@ func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64)
 			case <-svc.syncTicker.C:
 				svc.lock.Lock()
 				if svc.syncer != nil {
-					svc.sync(cancelCtx)
+					// If selective sync is not enabled (false), we default to not sync until the trigger
+					// condition has been checked. Otherwise, the default behavior is to sync.
+					shouldSync := !svc.selectiveSyncEnabled
+					// If selective sync is enabled and the sensor has been properly initialized,
```

Comment:
Since it'll only log once if we are unable to initialize a selective syncer, it'd be useful to continue to error log here that the sensor has not been properly initialized and that we are not syncing.

Reply:
We will log every time we reconfigure and try to reinitialize the syncer. I'm not sure if it's necessary here or would seem spamy if we output a log at every sync interval as well in addition to at reconfigure. What do you think?

Reply:
Can we expand on the "unable to initialize selective syncer" log? Perhaps `unable to initialize selective syncer; will not sync at all until fixed or removed from config`?

Reply:
Yep!